sunpy/map/map_factory.py

```diff
@@ -172,6 +179,10 @@ def _read_file(self, fname, **kwargs):
                 data = filedata
                 meta = MetaDict(filemeta)
                 new_pairs.append((data, meta))
+
+        if not new_pairs:
+            raise NoMapsInFileError(f"Reading the file '{fname}' found no HDUs with NAXIS > 1, so no maps could be loaded.")
```

Comment:
This isn't actually a check on the `NAXIS` keyword, but on the shape of the data, so I think the check should be changed or the warning should be changed to reflect the check (that the data has >= two dimensions).